strr-host-pm-web/app/stores/hostApplication.ts

```diff
@@ -1,55 +1,96 @@
 
 export const useHostApplicationStore = defineStore('host/application', () => {
-  // const { t } = useI18n()
-
-  const residenceSchema = z.object({
-    isPrincipalResidence: z.boolean(),
-    agreedToRentalAct: z.boolean(),
-    nonPrincipalOption: optionalOrEmptyString,
-    specifiedServiceProvider: optionalOrEmptyString,
-    agreedToSubmit: z.boolean()
+  const { t } = useI18n()
+  const { postApplication } = useStrrApi()
+
+  const propertyStore = useHostPropertyStore()
+  const reqStore = usePropertyReqStore()
+  const ownerStore = useHostOwnerStore()
+  const documentStore = useDocumentStore()
+  const confirmationSchema = z.object({
+    agreedToRentalAct: z.boolean().refine(val => val, { message: t('validation.required') }),
+    agreedToSubmit: z.boolean().refine(val => val, { message: t('validation.required') })
   })
 
-  const getEmptyResidenceDetails = () => ({
-    isPrincipalResidence: undefined,
-    agreedToRentalAct: undefined,
-    nonPrincipalOption: '',
-    specifiedServiceProvider: '',
-    agreedToSubmit: undefined
+  const getEmptyConfirmation = () => ({
+    agreedToRentalAct: false,
+    agreedToSubmit: false
   })
 
-  const residenceDetails = ref<ApiResidence>(getEmptyResidenceDetails())
-  const documents = ref<ApiDocument[]>([])
+  const userConfirmation = ref(getEmptyConfirmation())
+
+  const validateUserConfirmation = (returnBool = false): MultiFormValidationResult | boolean => {
+    const result = validateSchemaAgainstState(confirmationSchema, userConfirmation.value, 'confirmation-form')
+
+    if (returnBool) {
+      return result.success === true
+    } else {
+      return [result]
+    }
+  }
+
+  const createApplicationBody = (): HostApplicationPayload => {
+    const host = ownerStore.findByRole(OwnerRole.HOST)
+    const cohost = ownerStore.findByRole(OwnerRole.CO_HOST)
+    const propertyManger = ownerStore.findByRole(OwnerRole.PROPERTY_MANAGER)
 
-  // TODO: add document schema?
-  // TODO: add document - takes file upload, post to api, push resp to documents ref
-  const removeDocumentAtIndex = (index: number) => {
-    documents.value.splice(index, 1)
+    return {
+      registration: {
+        registrationType: ApplicationType.HOST,
+        primaryContact: formatOwnerHostAPI(host as HostOwner),
+        ...(cohost
+          ? { secondaryContact: formatOwnerHostAPI(cohost) }
+          : {}
+        ),
+        ...(propertyManger
+          ? { propertyManager: formatOwnerPropertyManagerAPI(propertyManger) }
+          : {}
+        ),
+        unitAddress: formatHostUnitAddressApi(propertyStore.unitAddress.address),
+        unitDetails: formatHostUnitDetailsAPI(propertyStore.unitDetails, propertyStore.blInfo),
+        documents: documentStore.apiDocuments,
+        principalResidence: {
+          // TODO: confirm mapping - assuming this is if PR is required or not?
+          isPrincipalResidence: !reqStore.prRequirements.isPropertyPrExempt,
```

Comment:
Would need to confirm but I think this is the rental unit setup type. rentalUnitSetupType === RentalUnitSetupType.WHOLE_PRINCIPAL_RESIDENCE

I'm just not sure if it needs to be the whole unit or if a room in the pr is okay too.

Reply:
I'll confirm this with lekshmi today

Reply:
haven't been able to get ahold of her so I'm going to merge and then will update as needed